provisioning_client/tests/hsm_client_riot_ut/hsm_client_riot_ut.c

```diff
@@ -250,6 +299,36 @@ static int my_DERtoPEM(DERBuilderContext* Context, uint32_t Type, char* PEM, uin
     return 0;
 }
 
+static void my_mbedtls_mpi_free(mbedtls_mpi* X)
+{
+    if (X == NULL)
```

Comment:
I need these to mirror exactly what the mbedtls API freeing functions are doing from the testing perspective, since those APIs are not mockable, if the pointer passed in is NULL we don't need to do anything

Reply:
In my mind we should know about attempts of double-free. I don't get why we mock these functions otherwise.

Reply:
Not sure I understand, or I'm not explaining it right
```C
void mbedtls_mpi_free( mbedtls_mpi *X )
{
    if( X == NULL )
        return;

    if( X->p != NULL )
    {
        mbedtls_mpi_zeroize( X->p, X->n );
        mbedtls_free( X->p );
    }

    X->s = 1;
    X->n = 0;
    X->p = NULL;
}
```
If the struct passed in is NULL, doesn't exist somehow we leave the function, if X->p is NULL/already been freed, we don't do anything with it except NULL it out again. Or are you saying that in the testing environment if we hit this call with a NULL pointer to log it? If that's the case then yes I can add a logging statement.

Reply:
I get what your asking now, a statement could be useful